libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -87,6 +87,7 @@ void AP_BattMonitor_UAVCAN::handle_battery_info(const BattInfoCb &cb)
     _interim_state.temperature = cb.msg->temperature;
     _interim_state.voltage = cb.msg->voltage;
     _interim_state.current_amps = cb.msg->current;
+    _soc = cb.msg->state_of_charge_pct;
```

Comment:
we need to cope with UAVCAN battery monitors that don't know the battery capacity. So either a setting, or cope with zero?

Reply:
If the state_of_charge_pct of UAVCAN BatteryInfo is always 0, use AP_BattMonitor_Backend::capacity_remaining_pct().

Reply:
@tridge Do we? The DSDL says that the field is required...

Reply:
@tridge standard says one thing, actual devices may say do something else, because life is like that.  Still, perhaps we should assume the field until we do find a counterexample.